src/tools/ceph_monstore_tool.cc

```diff
@@ -701,6 +701,87 @@ int rebuild_monstore(const char* progname,
   return 0;
 }
 
+void get_name_by_pid(pid_t pid, char *task_name)
+{
+  char proc_pid_path[PATH_MAX] = {0};
+  snprintf(proc_pid_path, PATH_MAX, "/proc/%d/cmdline", pid);
+  int fd = open(proc_pid_path, O_RDONLY);
+
+  if (fd < 0) {
+    fd = -errno;
+    std::cerr << "Failed to open '" << proc_pid_path
+              << "' error = " << cpp_strerror(fd)
+              << std::endl;
+    return;
+  }
+
+  int ret = read(fd, task_name, PATH_MAX-1);
+  if (ret < 0) {
+    ret = -errno;
+    std::cerr << "Failed to read '" << proc_pid_path
+              << "' error = " << cpp_strerror(ret)
+              << std::endl;
+    close(fd);
+  }
+
+  close(fd);
+}
+
+
+int check_lock(const string &path) {
+ 
+  string::const_reverse_iterator rit;
+  int pos = 0;
+  for (rit = path.rbegin(); rit != path.rend(); ++rit, ++pos) {
+    if (*rit != '/')
+      break;
+  }
+  ostringstream os;
+  os << path.substr(0, path.size() - pos) << "/store.db/LOCK";
+  string lock_path = os.str();
+
+  //The lock file is not existed.
+  if (access(lock_path.c_str(), F_OK)) {
+     return 0;
+  }
+
+  int fd = open(lock_path.c_str(), O_RDONLY);
+  if (fd < 0) {
+    fd = -errno;
+    std::cerr << "Failed to open '" << lock_path
+              << "' error = " << cpp_strerror(fd)
+              << std::endl;
+    return -errno;
+  }
+
+  struct flock f;
+  memset(&f, 0, sizeof(f));
+  f.l_type = F_WRLCK;
+  f.l_whence = SEEK_SET;
+
+  if (-1 == fcntl(fd, F_GETLK, &f)) {
+    std::cerr << " Failed to get the lock " << cpp_strerror(errno) << std::endl;
+    close(fd);
+    return -errno;
+  }
+
+  if (f.l_type != F_UNLCK) {
+    if (f.l_type == F_WRLCK) {
+      char task_name[PATH_MAX] = "unknown";
+      //get the locked thread name
+      get_name_by_pid(f.l_pid, task_name);
```

Comment:
@tchaikov  But I think it is would be better for user who use this tool. when the tool can not use, he can very quickly and clearly know which process handler the lock.and save his time to find the process.